aspnet/fundamentals/app-state.rst

```diff
@@ -1,8 +1,252 @@
+	:local:
+	:depth: 1
+
+`Download sample from GitHub <https://github.com/aspnet/docs/aspnet/fundamentals/app-state/sample>`_. 
+
+Application State Options
+-------------------------
+`Application state` refers to any data that is used to represent the current representation of the application. This includes both global and user-specific data. Previous versions of ASP.NET (and even ASP) have had built-in support for global ``Application`` and ``Session`` state stores, as well as a variety of other options. 
+
+.. note:: The ``Application`` store had the same characteristics as the ASP.NET ``Cache``, with fewer capabilities. In ASP.NET 5, ``Application`` no longer exists; applications written for previous versions of ASP.NET that are migrating to ASP.NET 5 replace ``Application`` with a :doc:`caching` implementation.
+
+Application developers are free to use different state storage providers depending on a variety of factors:
+
+- How long does the data need to persist?
+- How large is the data?
+- What format is the data?
+- Can it be serialized?
+- How sensitive was the data? Could it be stored on the client?
+
+Based on answers to these questions, application state in ASP.NET 5 apps can be stored or managed in a variety of ways.
+
+HttpContext.Items
+^^^^^^^^^^^^^^^^^
+The ``Items`` collection is the best location to store data that is only needed while processing a given request. Its contents are disposed of with each request. It is best used as a means of communicating between components or middleware that operate at different points in time during a request, and have no direct relationship with one another through which to pass parameters or return values. See `Working with HttpContext.Items`_, below.
+
+Querystring and Post
+^^^^^^^^^^^^^^^^^^^^
+State from one request can be provided to another request by adding values to the new request's querystring or by POSTing the data. This requires that the data be sent to the client and then sent back to the server, so it is best for small amounts of data. Querystrings are especially useful for capturing state in a persistent manner, allowing links with embedded state to be created and sent via email or social networks, for use potentially far into the future.
+
+Cookies
+^^^^^^^
+Very small pieces of state-related data can be stored in Cookies. These are sent with every request, and so the size should be kept to a minimum. Ideally, only an identifier should be used, with the actual data stored somewhere on the server, keyed to the identifier.
+
+Session
+^^^^^^^
+Session storage relies on a cookie-based identifier to access data related to a given user session (a series of requests from a particular browser and machine). It is a good place to store application state that is specific to a particular user but which doesn't need to be persisted permanently (or which can be reproduced as needed from a persistent store). See `Installing and Configuring Session`_, below for more details.
```

Comment:
"related to a given user session" -> browser session. You can't assume a session is restricted to a single user. Again you must be careful with storing sensitive information here.